pkg/commonconfig/commonconfig.go

```diff
@@ -0,0 +1,65 @@
+package commonconfig
+
+import (
+	"strings"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	dataClient "github.com/giantswarm/kubectl-gs/pkg/data/client"
+
+	"github.com/giantswarm/kubectl-gs/internal/key"
+)
+
+type CommonConfig struct {
+	configFlags genericclioptions.RESTClientGetter
+}
+
+func New(cf genericclioptions.RESTClientGetter) *CommonConfig {
+	cc := &CommonConfig{
+		configFlags: cf,
+	}
+
+	return cc
+}
+
+func (cc *CommonConfig) GetProvider() (string, error) {
+	config, err := cc.configFlags.ToRESTConfig()
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	var provider string
+	switch {
+	case strings.Contains(config.Host, key.ProviderAWS):
```

Comment:
Might we run into problems when/if we have an installation called `outlaws` or `jigsaws` that happen to be non AWS installations 😂 

https://www.thefreedictionary.com/words-containing-aws

Reply:
Good call! 👍 Improved the logic and added some tests for this